src/items.cpp

```diff
@@ -1224,6 +1228,14 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
 							}
 						}
 
+						// datapack compatibility, presume damage to be initialdamage if initialdamage is not declared.
+						// To avoid any initialdamage, add initialdamage xml property and set it to 0.
+						if (!initdamageConfigured && damage != 0) {
```

Comment:
here simply check if initDamage != 0 and set initDamage else set damage, damage is mandatory for field to work, so it should never be 0 in this case

Reply:
No, because initdamage can be configured to 0 in order to avoid any initial damage. (else damage will be initialdamage). 
Example if you want a field to intentionally begin a burn condition, but don't want it to initially deal any damage. 

The bool figures out if the 0 value is from code initialization, or loaded from ItemNode. This is critical to determine if initialdamage should be overriden with damage. I originally intended to initialize initdamage to null, but I don't think int32_t support null initializations, so I supplemented it with a bool.

Reply:
@Znote before my changes, you didn't have this option, so why do you have it now?

Reply:
You can just set initDamage to -1 to check if it was parsed from xml file instead of 0 and this bool is redundant then

Reply:
Ohh, I though you meant initDamage. damage is **not** mandatory as you say, there are fields (like small fire fields) that are completely without any damage. Thus a check for damage != 0 is neccesary. (I think)

Reply:
> You can just set initDamage to -1 to check if it was parsed from xml file instead of 0 and this bool is redundant then

No because `<attribute key="initdamage" value="1" />` will translate to -1 in the source. :P (-1 reduction to hit points). 
Theoretically, the opposite is true for a field that heals you. (healing field?). So I shouldn't twist the logic either.

Reply:
no it wont convert, just save the value raw and don't cast it to negative later when you have the code to use initDamage variable just use -initDamage

Reply:
Aight, done.